### PR TITLE
Move resources to a shared libary

### PR DESCRIPTION
**Pull request by TheOneRing**
This allows us to use resources in the cmd app and the unit tests.

This became relevant with https://github.com/owncloud/client/pull/9813

Fixes: #8839

### CODE REVIEW

File: src/resources/loadresources.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#include "loadresources.h"
+
+#include <qglobal.h>
+
+using namespace OCC;
+
+void static load_rc()
+{
+    Q_INIT_RESOURCE(ownCloudResources_translations);
```
Comment (fmoc) [inline src/resources/loadresources.cpp:9]: According to the Qt docs, we should not need any of these calls when using a shared library, it should be handled by Qt itself. I assume you have tested this without these calls? I do not have any experience moving Qt resources into a dedicated shared library, so I'm curious.

https://doc.qt.io/qt-6/resources.html

File: src/resources/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+target_link_libraries(ownCloudResources PUBLIC Qt5::Core)
+
+set_target_properties(ownCloudResources PROPERTIES
+    OUTPUT_NAME "${APPLICATION_EXECUTABLE}Resources"
+    AUTOUIC ON
+    AUTORCC ON
+)
+
+GENERATE_EXPORT_HEADER(ownCloudResources
```
Comment (fmoc) [inline src/resources/CMakeLists.txt:19]:
```suggestion
generate_export_header(ownCloudResources
```

File: src/resources/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        EXPORT_MACRO_NAME OWNCLOUDRESOURCES_EXPORT
+        EXPORT_FILE_NAME owncloudresources.h
+        STATIC_DEFINE OWNCLOUD_BUILT_AS_STATIC
```
Comment (fmoc) [inline src/resources/CMakeLists.txt:22]: This define is never used.

File: src/gui/CMakeLists.txt [hunk 4]
```diff
@@ -186,28 +186,8 @@ set_target_properties(owncloud PROPERTIES
...
-        PREFIX translations
-        FILES "${client_compiled_translations}"
-    )
-endif()
-
-#TODO Move resources files
-target_sources(owncloud PRIVATE
-    ${PROJECT_SOURCE_DIR}/client.qrc
-    ${PROJECT_SOURCE_DIR}/core_theme.qrc)
+target_link_libraries(owncloud owncloudCore ownCloudResources)
```
Comment (fmoc) [inline src/gui/CMakeLists.txt:189]: Why not link the resources into the core library?

File: src/resources/loadresources.cpp (hunk 1 above)
Comment (TheOneRing) [reply src/resources/loadresources.cpp:9]: Hm I haven't tested it, however we would still need to export a and use symbols from this lib.

File: src/resources/CMakeLists.txt (hunk 3 above)
Comment (TheOneRing) [reply src/resources/CMakeLists.txt:22]: Its used in the generated export header.

File: src/gui/main.cpp [hunk 5]
```diff
@@ -41,8 +42,8 @@ void warnSystray()
 
 int main(int argc, char **argv)
 {
-    Q_INIT_RESOURCE(client);
-
+    // load the resources
+    const OCC::ResourcesLoader resource;
```
Comment (fmoc) [inline src/gui/main.cpp:46]: This variable is called `resources` in the test utils. You might want to keep the naming consistent.

**Review by fmoc — APPROVED** [review]

**Pull request merged**